### PR TITLE
RSpec tests for Enumerable Numbers

### PR DESCRIPTION
**Pull request by bruna-genz**
In this branch, I created test cases for six methods of my enumerable numbers project. There are some tests that have failed, exposing that I didn't consider all possible scenarios for some of my methods.

### CODE REVIEW

**Comment by bruna-genz** [comment]
The linter errors are coming from the spec_helper.rb file.

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    it 'returns the original array' do
+      output = []
+      array.my_each { |element| output << element }
+      expect(output).to eql([1, 2, 3, 4])
+    end
+
+    it 'returns an enumerable if no code block is given' do
+      expect(array.my_each).to be_a(Enumerable)
+    end
```
Comment (samgaco) [inline spec/enumerables_spec.rb:15]: some tests are failing when running rspec

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+require './lib/enumerable_number'
+
+RSpec.describe Enumerable do
+  describe 'my_each' do
+    let(:array) { [1, 2, 3, 4] }
+
+    it 'returns the original array' do
+      output = []
+      array.my_each { |element| output << element }
```
Comment (samgaco) [inline spec/enumerables_spec.rb]: all the specs should be wrapped within describe blocks, [reference](https://github.com/rubocop-hq/rspec-style-guide#empty-lines-around-examples).

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @bruna-genz :high_brightness: 

Some tests seem to be failing when running rspec. Should be addressed.

Please submit back once you are done.


reference:
```

Failures:

  1) Enumerable my_each returns an enumerable if no code block is given
     Failure/Error: yield(self[i])
     
     LocalJumpError:
       no block given (yield)
     # ./lib/enumerable_number.rb:7:in `my_each'
     # ./spec/enumerables_spec.rb:16:in `block (3 levels) in <top (required)>'

  2) Enumerable my_select returns an enumerable if no code block is given
     Failure/Error: result << element if yield(element)
     
     LocalJumpError:
       no block given (yield)
     # ./lib/enumerable_number.rb:23:in `block in my_select'
     # ./lib/enumerable_number.rb:7:in `my_each'
     # ./lib/enumerable_number.rb:22:in `my_select'
     # ./spec/enumerables_spec.rb:36:in `block (3 levels) in <top (required)>'

  3) Enumerable my_any? when an argument is given returns true if any of the elements is === to the argument
     Failure/Error:
       def my_any?
         result = false
         if block_given?
           my_each do |element|
             result = true if yield(element)
           end
         else
           my_each do |element|
             result = true if element
           end
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
     # ./lib/enumerable_number.rb:42:in `my_any?'
     # ./spec/enumerables_spec.rb:93:in `block (4 levels) in <top (required)>'

  4) Enumerable my_inject when the operator is passed as a symbol without other argument passes each element and accumulated each result sequentially
     Failure/Error: operation = arg2.to_proc
     
     NoMethodError:
       undefined method `to_proc' for nil:NilClass
     # ./lib/enumerable_number.rb:107:in `my_inject'
     # ./spec/enumerables_spec.rb:149:in `block (4 levels) in <top (required)>'

Finished in 0.00636 seconds (files took 0.08046 seconds to load)
24 examples, 4 failures
```


-----------------

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Review by bolah2009 — APPROVED** [review]
## Nice job implementing all suggested changes.

### It's a :+1: from me,

You can merge to master. 

#### I strongly recommend you `fix` the linter warnings before merging. You can do this by running `rubocop -a`.

**Pull request merged**